docs(metric-applicability): split primary vs other, group others by family

`docs/reference/metric-applicability.md` lists 38 metrics in one flat "Cross-metric matrix" with a `Canonical role` column that mixes "Cell-canonical (IC)", "HAC variant of ic", "Series-tools", etc. Readers asking "what other metrics apply to cell X" must scan every row; family membership is implied by prose alone.

## Constraints

- `ts_beta` is the cell-canonical for both Common × Continuous and Common × Sparse (dummy factor) — one metric, two cells.
- `spanning_alpha`, `greedy_forward_selection` are spread-series consumers; `hit_rate`, `ic_trend`, `multi_split_oos_decay` are series-tools — not bound to any `(scope, signal)` cell.
- Merged-cell tables via HTML `rowspan` render in mkdocs-material but produce brittle diffs and complicate drift guards.

## Target structure

The page already opens with the cell-canonical primary table (the `_generated_evaluate_metric_table.md` include under `## Cell to evaluate-metric`). The remaining work restructures the cross-metric matrix lower on the page.

The flat 38-row table splits into family-grouped H3 subsections (no rowspan), each carrying cell context in its heading:

- IC family (`ic_newey_west`, `ic_ir`, `regime_ic`, `multi_horizon_ic`) — Cell: IND × CONT
- FM family (`pooled_ols`, `beta_sign_consistency`) — Cell: IND × CONT
- Quantile / Monotonicity / Concentration (`quantile_spread`, `quantile_spread_vw`, `monotonicity`, `top_concentration`) — Cell: IND × CONT
- Tradability (`turnover`, `notional_turnover`, `breakeven_cost`, `net_spread`) — Cell: IND × CONT
- Event-quality (`bmp_test`, `event_hit_rate`, `event_ic`, `profit_factor`, `event_skewness`, `signal_density`, `event_around_return`, `multi_horizon_hit_rate`, `mfe_mae_summary`, `clustering_diagnostic`, `corrado_rank_test`) — Cell: IND × SPARSE
- TS-β family (`mean_r_squared`, `compute_rolling_mean_beta`, `ts_beta_sign_consistency`, `ts_quantile_spread`, `ts_asymmetry`) — Cell: COMMON × CONT
- Spread-series consumers (`spanning_alpha`, `greedy_forward_selection`) — not cell-bound
- Series-tools (`hit_rate`, `ic_trend`, `multi_split_oos_decay`) — not cell-bound

Cell-canonical primaries (`ic`, `fama_macbeth`, `caar`, `ts_beta`) are excluded from the family subsections — they are the SSOT in the primary table at the top.

Each subsection retains the per-row schema (sample axis, min sample) but drops the `Cell` and `Canonical role` columns — both are now expressed in the section heading.

## Definition of Done

- [ ] Cross-metric matrix restructured into family-grouped H3 subsections.
- [ ] `Cell` and `Canonical role` columns dropped from the per-row schema.
- [ ] No drift guard regressions — primary metric set remains the single source from `_procedures.py`; family grouping is editorial only.
- [ ] `tests/test_list_metrics.py::_parse_applicability_doc` regex updated to walk the new `### Family — Cell: <name>` section schema; all 4 `(scope, signal)` cells parse at least one metric row (assertion `assert expected` does not fire).

Refs #130, #139, #140, #144

Reopening: DoD #3 ("No drift guard regressions") was not actually verified at merge time.

`tests/test_list_metrics.py::_parse_applicability_doc` regex (`^\|\s*\[`([^`]+)`\][^|]*\|\s*([^|]+?)\s*\|`) assumes the old flat schema where the second column carried the cell name (`Individual × Continuous` etc.). After the family-grouped restructure (#151), the cell moved to the `### Family — Cell: ...` section heading and the second column is now `Sample axis` (`T`, `T per regime`, ...). Parser silently falls through every cell branch and returns 0 rows for all 4 `(scope, signal)` cells; the assertion `assert expected, "...found no rows for this cell"` fires on main and on every PR.

Confirmed pre-existing on main:
```
FAILED tests/test_list_metrics.py::test_list_metrics_matches_applicability_doc[individual-continuous]
FAILED tests/test_list_metrics.py::test_list_metrics_matches_applicability_doc[common-continuous]
FAILED tests/test_list_metrics.py::test_list_metrics_matches_applicability_doc[individual-sparse]
FAILED tests/test_list_metrics.py::test_list_metrics_matches_applicability_doc[common-sparse]
```

Adding one DoD bullet about parser/schema alignment so the close criterion is verifiable next time.

--- a/tests/test_list_metrics.py
+++ b/tests/test_list_metrics.py
@@ -22,41 +22,90 @@ from factrix._metric_index import _STAGE1_HELPERS, MetricRow, user_facing_rows
 _APPLICABILITY_DOC = pathlib.Path("docs/reference/metric-applicability.md")
 
 
+# Cell-canonical primaries: SSOT moved to the auto-generated dispatch
+# table at the top of metric-applicability.md (#142), whose tuple keys
+# are dispatch-keyed and broader than authoring scope (e.g. ts_beta
+# dispatches on (COMMON, SPARSE, *, PANEL) but its Matrix-row authoring
+# cell is (COMMON, CONTINUOUS, *, PANEL); list_metrics returns the
+# authoring set). Inject primaries by their authoring cell rather than
+# parse the dispatch table — adding a primary requires updating one
+# literal here, the same cadence as updating Matrix-row tags.
+_PRIMARY_METRIC_CELLS: dict[str, list[tuple[FactorScope, Signal]]] = {
+    "ic": [(FactorScope.INDIVIDUAL, Signal.CONTINUOUS)],
+    "fama_macbeth": [(FactorScope.INDIVIDUAL, Signal.CONTINUOUS)],
+    "caar": [
+        (FactorScope.INDIVIDUAL, Signal.SPARSE),
+        (FactorScope.COMMON, Signal.SPARSE),
+    ],
+    "ts_beta": [(FactorScope.COMMON, Signal.CONTINUOUS)],
+}
+
+# Family-subsection cell-name → (scope, signal) cells.
+_CELL_HEADING_MAP: dict[str, list[tuple[FactorScope, Signal]]] = {
+    "Individual × Continuous": [(FactorScope.INDIVIDUAL, Signal.CONTINUOUS)],
+    "Common × Continuous": [(FactorScope.COMMON, Signal.CONTINUOUS)],
+    # Matrix-row tags use ``(*, SPARSE, *, PANEL)``: wildcard scope.
+    "Individual × Sparse": [
+        (FactorScope.INDIVIDUAL, Signal.SPARSE),
+        (FactorScope.COMMON, Signal.SPARSE),
+    ],
+    "Common × Sparse": [(FactorScope.COMMON, Signal.SPARSE)],
+}
+
+# Not-cell-bound family-name → cells. Spread-series consumers operate on
+# the Individual × Continuous quantile spread output; series-tools are
+# wildcard-scope, ``(*, CONTINUOUS, *, TIMESERIES)``.
+_NOT_CELL_BOUND_MAP: dict[str, list[tuple[FactorScope, Signal]]] = {
+    "Spread-series consumers": [(FactorScope.INDIVIDUAL, Signal.CONTINUOUS)],
+    "Series-tools": [
+        (FactorScope.INDIVIDUAL, Signal.CONTINUOUS),
+        (FactorScope.COMMON, Signal.CONTINUOUS),
+    ],
+}
+
+
 def _parse_applicability_doc() -> dict[tuple[FactorScope, Signal], set[str]]:
-    """Build per-cell expected name sets from the applicability matrix."""
+    """Build per-cell expected name sets from the applicability matrix.
+
+    Walks the family subsections under ``## Other metrics by family``:
+    each ``### <Family> — Cell: <CellName>`` heading sets the active
+    cell list; subsequent ``| [`name`]...`` table rows are assigned to
+    it. ``### <Family> — not cell-bound`` activates the wildcard
+    families (spread-series / series-tools). Cell-canonical primaries
+    are injected from :data:`_PRIMARY_METRIC_CELLS`. Each new ``##``
+    heading clears the active list so rows in unrelated sections
+    (event-study contracts, sample-size constants) are ignored.
+    """
     expected: dict[tuple[FactorScope, Signal], set[str]] = {
         (FactorScope.INDIVIDUAL, Signal.CONTINUOUS): set(),
         (FactorScope.COMMON, Signal.CONTINUOUS): set(),
         (FactorScope.INDIVIDUAL, Signal.SPARSE): set(),
         (FactorScope.COMMON, Signal.SPARSE): set(),
     }
+    for name, cells in _PRIMARY_METRIC_CELLS.items():
+        for cell in cells:
+            expected[cell].add(name)
+
     text = _APPLICABILITY_DOC.read_text(encoding="utf-8")
-    row_re = re.compile(r"^\|\s*\[`([^`]+)`\][^|]*\|\s*([^|]+?)\s*\|", re.MULTILINE)
-    for match in row_re.finditer(text):
-        name, cell = match.group(1), match.group(2)
-        # Drop the leading symbol on the metric name when present
-        # (table syntax already strips backticks).
-        targets: list[tuple[FactorScope, Signal]] = []
-        if cell.startswith("Individual × Continuous") or cell.startswith(
-            "Spread-series consumer"
-        ):
-            targets = [(FactorScope.INDIVIDUAL, Signal.CONTINUOUS)]
-        elif cell.startswith("Common × Continuous"):
-            targets = [(FactorScope.COMMON, Signal.CONTINUOUS)]
-        elif cell.startswith("Individual × Sparse"):
-            # Matrix-row tags use ``(*, SPARSE, *, PANEL)``: wildcard scope.
-            targets = [
-                (FactorScope.INDIVIDUAL, Signal.SPARSE),
-                (FactorScope.COMMON, Signal.SPARSE),
-            ]
-        elif cell.startswith("Series-tools"):
-            # ``(*, CONTINUOUS, *, TIMESERIES)``: wildcard scope.
-            targets = [
-                (FactorScope.INDIVIDUAL, Signal.CONTINUOUS),
-                (FactorScope.COMMON, Signal.CONTINUOUS),
-            ]
-        for t in targets:
-            expected[t].add(name)
+    cell_re = re.compile(r"^###\s+(.+?)\s+—\s+Cell:\s+(.+?)\s*$")
+    nocell_re = re.compile(r"^###\s+(.+?)\s+—\s+not cell-bound\s*$")
+    row_re = re.compile(r"^\|\s*\[`([^`]+)`\]")
+    h2_re = re.compile(r"^##\s")
+
+    active: list[tuple[FactorScope, Signal]] = []
+    for line in text.splitlines():
+        if h2_re.match(line):
+            active = []
+            continue
+        if m := cell_re.match(line):
+            active = _CELL_HEADING_MAP.get(m.group(2).strip(), [])
+            continue
+        if m := nocell_re.match(line):
+            active = _NOT_CELL_BOUND_MAP.get(m.group(1).strip(), [])
+            continue
+        if active and (m := row_re.match(line)):
+            for cell in active:
+                expected[cell].add(m.group(1))
     return expected
 
 

